lib/puppet/util/windows/process.rb

```diff
@@ -351,4 +382,32 @@ class TOKEN_ELEVATION < FFI::Struct
   ffi_lib :advapi32
   attach_function_private :GetTokenInformation,
     [:handle, TOKEN_INFORMATION_CLASS, :lpvoid, :dword, :pdword ], :win32_bool
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/ms724834%28v=vs.85%29.aspx
+  # typedef struct _OSVERSIONINFO {
+  #   DWORD dwOSVersionInfoSize;
+  #   DWORD dwMajorVersion;
+  #   DWORD dwMinorVersion;
+  #   DWORD dwBuildNumber;
+  #   DWORD dwPlatformId;
+  #   TCHAR szCSDVersion[128];
+  # } OSVERSIONINFO;
+  class OSVERSIONINFO < FFI::Struct
+    layout(
+      :dwOSVersionInfoSize, :dword,
+      :dwMajorVersion, :dword,
+      :dwMinorVersion, :dword,
+      :dwBuildNumber, :dword,
+      :dwPlatformId, :dword,
+      :szCSDVersion, [:wchar, 128]
+    )
+  end
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/ms724451(v=vs.85).aspx
+  # BOOL WINAPI GetVersionEx(
+  #   _Inout_  LPOSVERSIONINFO lpVersionInfo
+  # );
+  ffi_lib :kernel32
+  attach_function_private :GetVersionExW,
```

Comment:
I thought about yanking this out into `os.rb` or similar, but given that would require more touches, I moved away from that solution.